src/libraries/Microsoft.PowerFx.Interpreter/Environment/SymbolExtensions.cs

```diff
@@ -31,6 +32,16 @@ public static void SetCulture(this RuntimeConfig symbols, CultureInfo culture)
             symbols.AddService(culture ?? throw new ArgumentNullException(nameof(culture)));
         }
 
+        /// <summary>
+        /// Set CancellationToken.
+        /// </summary>
+        /// <param name="symbols">SymbolValues where to set the CancellationToken.</param>
+        /// <param name="cancellationToken">CancellationToken to set.</param>
+        public static void SetCancellationToken(this RuntimeConfig symbols, CancellationToken cancellationToken)
+        {
+            symbols.AddService(cancellationToken);
+        }
```

Comment:
Cancellation token shouldn't be a service. 

Most services are persistent and  could be used across multiple evals. 
Whereas a cancellation token is "consumed" per eval. So worth just passing as a separate parameter.

Reply:
Updated the code to pass this as separate parameter if defined.